src/disk.c

```diff
@@ -875,6 +873,30 @@ static int disk_read(void) {
 #endif
   } /* while (fgets (buffer, sizeof (buffer), fh) != NULL) */
 
+  /* Remove disks that have disappeared from diskstats */
+  for (ds = disklist, pre_ds = disklist; ds != NULL;) {
+    /* Disk exists */
+    if (ds->poll_count == poll_count) {
+      pre_ds = ds;
+      ds = ds->next;
+      continue;
+    }
+
+    /* Disk is missing, remove it */
+    diskstats_t *missing_ds = ds;
+    if (ds == disklist) {
+      disklist = ds->next;
+      ds = disklist;
+      pre_ds = ds;
```

Comment:
Optional: I find this use of `ds` a bit confusing. Suggestion:

```c
disklist = pre_ds = ds->next;
ds = disklist;
```

Reply:
I stopped at the following code:
```c
    /* Disk is missing, remove it */
    diskstats_t *missing_ds = ds;
    if (ds == disklist) {
      pre_ds = disklist = ds->next;
    } else {
      pre_ds->next = ds->next;
    }
    ds = ds->next;

```